File: src/phorge/api/endpoints/jobs.py
"""Scheduled job API endpoints."""

from __future__ import annotations

from typing import Any

from phorge.api.client import ForgeClient
from phorge.api.models import ScheduledJob


class ScheduledJobsAPI:
    def __init__(self, client: ForgeClient) -> None:
        self._client = client

    async def list(self, server_id: int) -> list[ScheduledJob]:
        data = await self._client.get(f"/servers/{server_id}/jobs")
        return [ScheduledJob(**j) for j in data.get("jobs", [])]

    async def get(self, server_id: int, job_id: int) -> ScheduledJob:
        data = await self._client.get(f"/servers/{server_id}/jobs/{job_id}")
        return ScheduledJob(**data["job"])

    async def create(
        self,
        server_id: int,
        *,
        command: str,
        frequency: str = "nightly",
        user: str = "forge",
    ) -> ScheduledJob:
        payload: dict[str, Any] = {
            "command": command,
            "frequency": frequency,
            "user": user,
        }
        data = await self._client.post(f"/servers/{server_id}/jobs", json=payload)
        return ScheduledJob(**data["job"])

    async def delete(self, server_id: int, job_id: int) -> None:
        await self._client.delete(f"/servers/{server_id}/jobs/{job_id}")

    async def get_output(self, server_id: int, job_id: int) -> str:
        data = await self._client.get(f"/servers/{server_id}/jobs/{job_id}/output")
        return data.get("output", "")

File: internal/forge/jobs.go
package forge

import (
	"context"
	"net/http"
)

// JobCreateOpts contains the options for creating a scheduled job.
type JobCreateOpts struct {
	Command   string `json:"command"`
	Frequency string `json:"frequency"` // default "nightly"
	User      string `json:"user"`      // default "forge"
	Name      string `json:"name,omitempty"`
}

// List returns all scheduled jobs on a server.
func (s *JobsService) List(ctx context.Context, serverID int64) ([]ScheduledJob, error) {
	return listResources[ScheduledJob](s.client, ctx, s.client.orgPath("/servers/%d/scheduled-jobs", serverID))
}

// Get returns a single scheduled job by ID.
func (s *JobsService) Get(ctx context.Context, serverID, jobID int64) (*ScheduledJob, error) {
	return getResource[ScheduledJob](s.client, ctx, s.client.orgPath("/servers/%d/scheduled-jobs/%d", serverID, jobID))
}

// Create creates a new scheduled job on a server.
func (s *JobsService) Create(ctx context.Context, serverID int64, opts JobCreateOpts) (*ScheduledJob, error) {
	return createResource[ScheduledJob](s.client, ctx, s.client.orgPath("/servers/%d/scheduled-jobs", serverID), opts)
}

// Delete removes a scheduled job from a server.
func (s *JobsService) Delete(ctx context.Context, serverID, jobID int64) error {
	return s.client.do(ctx, http.MethodDelete, s.client.orgPath("/servers/%d/scheduled-jobs/%d", serverID, jobID), nil, nil)
}
